lib/src/async-compiler.ts

```diff
@@ -6,42 +6,144 @@ import {spawn} from 'child_process';
 import {Observable} from 'rxjs';
 import {takeUntil} from 'rxjs/operators';
 
+import {
+  OptionsWithLegacy,
+  StringOptionsWithLegacy,
+  createDispatcher,
+  handleCompileResponse,
+  handleLogEvent,
+  newCompilePathRequest,
+  newCompileStringRequest,
+} from './compiler';
 import {compilerCommand} from './compiler-path';
+import {FunctionRegistry} from './function-registry';
+import {ImporterRegistry} from './importer-registry';
+import * as utils from './utils';
+import * as proto from './vendor/embedded_sass_pb';
+import {CompileResult} from './vendor/sass';
 
 /**
- * An asynchronous wrapper for the embedded Sass compiler that exposes its stdio
- * streams as Observables.
+ * An asynchronous wrapper for the embedded Sass compiler
  */
-export class AsyncEmbeddedCompiler {
+export class AsyncCompiler {
   /** The underlying process that's being wrapped. */
   private readonly process = spawn(
     compilerCommand[0],
     [...compilerCommand.slice(1), '--embedded'],
     {windowsHide: true}
   );
 
+  /** Whether the underlying compiler has already exited. */
+  private disposed = false;
+
+  /** A record of all compilations */
+  private compilations: Promise<CompileResult>[] = [];
+
   /** The child process's exit event. */
-  readonly exit$ = new Promise<number | null>(resolve => {
+  private readonly exit$ = new Promise<number | null>(resolve => {
     this.process.on('exit', code => resolve(code));
   });
 
   /** The buffers emitted by the child process's stdout. */
-  readonly stdout$ = new Observable<Buffer>(observer => {
+  private readonly stdout$ = new Observable<Buffer>(observer => {
     this.process.stdout.on('data', buffer => observer.next(buffer));
   }).pipe(takeUntil(this.exit$));
 
   /** The buffers emitted by the child process's stderr. */
-  readonly stderr$ = new Observable<Buffer>(observer => {
+  private readonly stderr$ = new Observable<Buffer>(observer => {
     this.process.stderr.on('data', buffer => observer.next(buffer));
   }).pipe(takeUntil(this.exit$));
 
   /** Writes `buffer` to the child process's stdin. */
-  writeStdin(buffer: Buffer): void {
+  private writeStdin(buffer: Buffer): void {
     this.process.stdin.write(buffer);
   }
 
+  private throwIfDisposed(): void {
```

Comment:
Document this.

Reply:
Fixed in cdd32ca12c739cf98367fc5c5294f8f96e2ed746